cpp/src/lance/io/reader.cc

```diff
@@ -141,7 +142,7 @@ Status FileReader::Open() {
   assert(metadata_->page_table_position() >= size - kPrefetchSize);
 
   auto num_batches = metadata_->num_batches();
-  auto num_columns = manifest_->schema()->GetFieldsCount();
+  auto num_columns = ranges::max(manifest_->schema()->GetFieldIds()) + 1;
```

Comment:
don't have to take care of it now but how would this work for column deletions?

Reply:
yea, so for now, we dont reclaim the deleted column Ids, so they are still exist in the lookup table, but do not exist in the schema. 

A simplest solution could be just run a new snapshot of everything and reclaim the column IDs.